### PR TITLE
Preparation to release v.1.9.0

### PR DESCRIPTION
**Pull request by Vladlukhanin**
*Make sure that you wrote the tests for your proposed changes and the existing test was success.*

**Made/Proposed changes:**
*(Don’t use general words, describe changes in details)*

- Refactored functionality of dialogs, chats and messages (optimized with backboneJS framework):
    * new logic for chat messages rendering,
    * new logic for chat messages' storage,
    * new logic to work with messages statuses;
- Bug fixes:
    * fixed some bugs with call's timer,
    * fixed attachments (if switch the chat during uploading),
    * corrected styles for Firefox browser (empty spaces, chat's input field, other),
    * optimized styles and animation for scroll,
    * cleaning of empty dialogs' section,
    * sync presence status between browser's tabs (or different browsers) from same user,
    * and fixed other minor bugs;
- Replaced the 'grunt-contrib-compass' on 'grunt-contrib-sass' (for faster compilation of styles).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 See [the Releases section](https://github.com/QuickBlox/q-municate-web/releases) for changelogs for each release version of Q-municate.
 
+## v. 1.9.0 - December 5, 2016
```
Comment (dimaspirit) [inline CHANGELOG.md]: Зачем ты копируешь changelog, если ты правильно ведёшь https://github.com/QuickBlox/q-municate-web/releases ? Может просто оставь ссылку

File: Gruntfile.js [hunk 2]
```diff
@@ -31,16 +31,13 @@ module.exports = function(grunt) {
...
+            dist: {
                 options: {
-                    cssDir: '<%= yeoman.app %>/.css',
-                    sassDir: '<%= yeoman.app %>/styles',
-                    javascriptsDir: '<%= yeoman.app %>/scripts',
-                    imagesDir: '<%= yeoman.app %>/images',
-                    noLineComments: true,
-                    relativeAssets: true,
-                    raw: 'preferred_syntax = :scss\n'
+                    style: 'expanded'
```
Comment (dimaspirit) [inline Gruntfile.js]: What are you using 'expanded' ? Are you minify then your css?

File: app/scripts/entities.js [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+        },
+
+        // keep count for messages collection
+        keepCountOfMessages: function() {
+            var dialogId = this.models[0].get('dialog_id'),
+                dialog = entities.Collections.dialogs.get(dialogId),
+                unreadCount = dialog.get('unread_count');
+
+            if (
+                ((this.length > 20) && (unreadCount < 20)) ||
```
Comment (dimaspirit) [inline app/scripts/entities.js:125]: Leave a last 50 msgs and areate a property in config for this const.

File: app/scripts/entities.js [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+                    read: 1
+                }, function() {
+                    dialog.set('unread_count', '');
+                });
+            }
+        },
+
+        saveDraft: function() {
+            if (entities.active) {
+                var dialog = this.get(entities.active),
```
Comment (dimaspirit) [inline app/scripts/entities.js]: bad idea. Work with html in view

File: app/scripts/main.js [hunk 5]
```diff
@@ -85,7 +86,9 @@ requirejs.config({
         // helpers
         Helpers: 'helpers',
         // templates
-        QMHtml: 'qmhtml'
+        QMHtml: 'qmhtml',
+        // entities (for refactoring)
```
Comment (dimaspirit) [inline app/scripts/main.js]: ?

File: app/scripts/entities.js (hunk 3 above)
Comment (Vladlukhanin) [reply app/scripts/entities.js:125]: https://quickblox.atlassian.net/browse/QMW-826 - for next sprint

**Pull request merged**